processing/src/main/java/io/druid/segment/data/BitmapValues.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.segment.data;
+
+import it.unimi.dsi.fastutil.ints.IntIterator;
+import it.unimi.dsi.fastutil.ints.IntIterators;
+
+/**
+ * Doesn't extend {@link it.unimi.dsi.fastutil.ints.IntIterable} to avoid accidential for-each iteration with boxing.
+ */
+public interface BitmapValues
+{
+  BitmapValues EMPTY = new BitmapValues()
+  {
+    @Override
+    public int size()
+    {
+      return 0;
+    }
+
+    @Override
+    public IntIterator iterator()
+    {
+      return IntIterators.EMPTY_ITERATOR;
+    }
+  };
+
+  int size();
```

Comment:
It looks like the most common use for `size()` is as an `isEmpty()` check, but the `size()` method in some bitmap implementations is not straight forward. Would it make sense to get rid of `size()` and have a `isEmpty()` instead, or have an `isEmpty()` in addition to `size()`?

Reply:
oh wait, i was looking at `ImmutableBitmap::size`, so it might be outside the scope of this PR.

Reply:
https://github.com/druid-io/druid/pull/4820